setup_ssl.sh

```diff
@@ -12,63 +16,112 @@ test -z ${mainDomain} && echo "No domain specified. Exiting" && exit 1
+  ls -la /etc/letsencrypt/live/${domain}/privkey.pem
+  
+  # Check if certificate and key was generated and put it in the vhosts configs
+  if [ -L /etc/letsencrypt/live/${domain}/fullchain.pem -a -L /etc/letsencrypt/live/${domain}/privkey.pem ]; then
+      fullchain=$(readlink -f /etc/letsencrypt/live/${domain}/fullchain.pem)
+      privkey=$(readlink -f /etc/letsencrypt/live/${domain}/privkey.pem)
+      if [ -f ${fullchain} -a -f ${privkey} ]; then
+        echo OK
+      fi
+  fi
+}
 
-echo "All DNS records exist and are correct"
+# Processing ZED domain(s)
+if [ ${ZED_HTTPS} -eq 1 ];then
+  myDomains="os.${mainDomain}"
+  checkDig
+  for domain in $(echo ${myDomains}); do
+    checkDomain ${domain}
+  done
 
-cd /tmp
+  getCertificates "${myDomains}"
 
-repoDir=/tmp/certbot
-# Get certbot sources
-if [ -d ${repoDir}/.git ]; then
-    cd ${repoDir}
-    git pull
-else
-    git clone https://github.com/certbot/certbot.git ${repoDir}
+  if [ "$(checkCertificates ${mainDomain})" == "OK" ]; then
+    echo j2 /etc/nginx/conf.d/ssl/ssl.vhost-zed.conf.j2 > /etc/nginx/conf.d/vhost-zed.conf
+  else
+    echo FAIL
+  fi
 fi
 
-${repoDir}/letsencrypt-auto certonly --webroot -w /usr/share/nginx/html -d $(echo ${myDomains}|sed "s/ / -d /g") -m admin@${mainDomain} --agree-tos -n --expand
-
+# Processing YVES domain(s)
+if [ ${YVES_HTTPS} -eq 1 ];then
+  myDomains="${mainDomain} www.${mainDomain}"
+  checkDig
+  for domain in $(echo ${myDomains}); do
+    checkDomain ${domain}
+  done
+  getCertificates "${myDomains}"
+  if [ "$(checkCertificates ${mainDomain})" == "OK" ]; then
+    echo j2 /etc/nginx/conf.d/ssl/ssl.vhost-yves.conf.j2 > /etc/nginx/conf.d/vhost-yves.conf
+  else
+    echo FAIL
+  fi
+fi
 
-ls -la /etc/letsencrypt/live/${mainDomain}/fullchain.pem
-ls -la /etc/letsencrypt/live/${mainDomain}/privkey.pem
+# Processing GLUE domain(s)
+if [ ${GLUE_HTTPS} -eq 1 ];then
+  myDomains="glue.${mainDomain}"
+  checkDig
+  for domain in $(echo ${myDomains}); do
+    checkDomain ${domain}
+  done
 
-# Check if certificate and key was generated and put it in the vhosts configs
-if [ -L /etc/letsencrypt/live/${mainDomain}/fullchain.pem -a -L /etc/letsencrypt/live/${mainDomain}/privkey.pem ]; then
-    fullchain=$(readlink -f /etc/letsencrypt/live/${mainDomain}/fullchain.pem)
-    privkey=$(readlink -f /etc/letsencrypt/live/${mainDomain}/privkey.pem)
-    if [ -f ${fullchain} -a -f ${privkey} ]; then
-        j2 /etc/nginx/conf.d/ssl/ssl.vhost-yves.conf.j2 > /etc/nginx/conf.d/vhost-yves.conf
-        j2 /etc/nginx/conf.d/ssl/ssl.vhost-zed.conf.j2 > /etc/nginx/conf.d/vhost-zed.conf
-        #j2 /etc/nginx/conf.d/ssl/ssl.vhost-glue.conf.j2 > /etc/nginx/conf.d/vhost-glue.conf
+  getCertificates "${myDomains}"
 
-	supervisorctl restart nginx
-    fi
+  if [ "$(checkCertificates ${mainDomain})" == "OK" ]; then
+    echo j2 /etc/nginx/conf.d/ssl/ssl.vhost-glue.conf.j2 > /etc/nginx/conf.d/vhost-glue.conf
+  else
+    echo FAIL
```

Comment:
Please and more information in this FAIL description.

Reply:
Sure, actually I just forgot about this condition :)